common/djangoapps/student/management/commands/bulk_change_enrollment_csv.py

```diff
@@ -0,0 +1,115 @@
+        mode.
+        Could be used to update effected users by order
+        placement issues. If large number of students are effected
+        in different courses.
+        Similar to bulk_change_enrollment but uses the csv file
+        input format and can enroll students in multiple courses.
+
+        Example:
+            $ ... bulk_change_enrollment_csv csv_file_path
+        """
+
+    def add_arguments(self, parser):
+        """ Add argument to the command parser. """
+        parser.add_argument(
+            '--csv_file_path',
+            required=True,
+            help='Csv file path'
+        )
+
+    def handle(self, *args, **options):
+        """ Main handler for the command."""
+        file_path = options['csv_file_path']
+
+        if not path.isfile(file_path):
+            raise CommandError("File not found.")
+
+        with open(file_path) as csv_file:
+            course_key = None
+            user = None
+            file_reader = csv.DictReader(csv_file)
+            for row in file_reader:
+                try:
+                    course_key = CourseKey.from_string(row['course_id'])
+                except InvalidKeyError:
+                    logger.warning('Invalid or non-existent course id [{}]'.format(row['course_id']))
+
+                try:
+                    user = User.objects.get(username=row['user'])
+                except:
+                    logger.warning('Invalid or non-existent user [{}]'.format(row['user']))
+                if course_key and user:
+                    try:
+                        course_enrollment = CourseEnrollment.get_enrollment(user, course_key)
+                        #  If student is not enrolled in course enroll the student in free mode
+                        if not course_enrollment:
+                            # try to create a enroll user in default course enrollment mode in case of
+                            # professional it will break because of no default course mode.
+                            try:
+                                course_enrollment = CourseEnrollment.get_or_create_enrollment(user=user,
+                                                                                              course_key=course_key)
+                            except Exception:  # pylint: disable=broad-except
+                                # In case if no free mode is available.
+                                course_enrollment = None
+
+                        if course_enrollment:
+                            # if student already had a enrollment and its mode is same as the provided one
+                            if course_enrollment.mode == row['mode']:
+                                logger.info("Student [%s] is already enrolled in Course [%s] in mode [%s].", user.username,
+                                            course_key, course_enrollment.mode)
+                                # set the enrollment to active if its not already active.
```

Comment:
Is there an existing method in the CourseEnrollment class that does this?  We should use it to reduce code duplication.

Reply:
updated